apps/builder/app/shared/pages/use-switch-page.ts

```diff
@@ -0,0 +1,30 @@
+import { useNavigate } from "@remix-run/react";
+import type { Page } from "@webstudio-is/project-build";
+import { useProject } from "~/builder/shared/nano-states";
+import { builderPath } from "~/shared/router-utils";
+import { useAuthToken } from "~/shared/nano-states";
+import { useSubscribe } from "~/shared/pubsub";
+
+export const useSwitchPage = () => {
+  const [project] = useProject();
+  const [authToken] = useAuthToken();
+  const navigate = useNavigate();
+
+  return (pageId: "home" | Page["id"]) => {
```

Comment:
Since we can access stores from anywhere it could look like this. A little less abstraction around remix.
```
const navigate = useNavigate();
navigate(getPageUrl(pageId))
```

Reply:
How can I access store without a hook?